core/src/Cabana_Parallel.hpp

```diff
@@ -74,6 +74,54 @@ KOKKOS_FORCEINLINE_FUNCTION
     functor( t, std::forward<IndexTypes>( indices )..., reduce_val );
 }
 
+template <class ExecutionPolicy, class Functor>
+struct ParallelFor;
+
+template <class Functor, int VectorLength, class... Properties>
+struct ParallelFor<SimdPolicy<VectorLength, Properties...>, Functor>
+{
+    using simd_policy = SimdPolicy<VectorLength, Properties...>;
+    using team_policy = typename simd_policy::base_type;
+    using work_tag = typename team_policy::work_tag;
+    using index_type = typename team_policy::index_type;
+    using member_type = typename team_policy::member_type;
+
+    simd_policy exec_policy_;
+    Functor functor_;
+
+    ParallelFor( std::string label, simd_policy exec_policy, Functor functor )
+        : exec_policy_( std::move( exec_policy ) )
+        , functor_( std::move( functor ) )
+    {
+        if ( label.empty() )
+            Kokkos::parallel_for(
+                dynamic_cast<const team_policy&>( exec_policy_ ), *this );
+        else
+            Kokkos::parallel_for(
+                label, dynamic_cast<const team_policy&>( exec_policy_ ),
+                *this );
+    }
+
+    template <class WorkTag>
+    KOKKOS_FUNCTION std::enable_if_t<!std::is_void<WorkTag>::value &&
+                                     std::is_same<WorkTag, work_tag>::value>
+    operator()( WorkTag, member_type const& team ) const
```

Comment:
So this is just adding flexibility if you want a different tag than the `SimdPolicy`?

Reply:
@dalg24 walked me through this - resolving